staging/src/k8s.io/apimachinery/pkg/util/httpstream/spdy/roundtripper_test.go

```diff
@@ -319,7 +334,11 @@ func TestRoundTripAndNewConnection(t *testing.T) {
 			if err != nil {
 				t.Fatalf("error creating request: %s", err)
 			}
-			req, err := http.NewRequest("GET", server.URL, nil)
+
+			ctx, cancel := context.WithTimeout(context.Background(), timeout)
```

Comment:
is this a realistic use case? IIUIC the context will handle the whole request lifecycle, for these kind of long lived request doesn't seem possible to use timeouts without breaking anything

Reply:
I've described our use case here: https://github.com/kubernetes/kubernetes/pull/116412#issuecomment-1463651074

Using the context here isn't for a long lived request, it's only for the initial roundtrip and upgrading the connection. Using the context here does not influence any upcoming actual streaming.